### ISSUE TITLE
Kid-Friendly mode

### ISSUE THREAD

**Issue by Volte6** [GoMudEngine/GoMud#115]
# Rationale

I observed in a reddit thread a parent looking for a mud that is kid-safe... realistically this means strangers can't communicate with their kid, as its the only way to guarantee safety.

This is an aspirational goal, not part of any critical path.

# Goals

* `mute` - Cannot `say`, `emote`, `broadcast` etc (unless from admin/mod)
* `deafen` - Does not receive `say`, `emote`, `broadcast` etc (unless from admin/mod)

* Add a bad-word filter that will apply to:
  * Names
  * Descriptions
  * `say` commands
  * `emote` commands
* Add a server config `SafetyMode`
  * `None` - no limitations
  * `Medium` - (Requires an age provided to logins) - Under 14 are set to `mute`/`deafen` & Descriptions are assigned/selected instead of freeform.
  * `High` - Everyone is set to `mute`/`deafen` & Descriptions are assigned/selected instead of freeform.
* Possible support general `mute` and `deafen` ad-hoc. This would also help address general abusers.

**Comment by Volte6** [comment]
mute and deafen added in this PR: https://github.com/Volte6/GoMud/pull/140

**Comment by Jasrags** [comment]
I like the idea of supporting age-based restrictions, but the main challenge is verifying age in a reliable way. If we’re targeting scenarios where the player is under 13 and doesn’t have their own email address, then self-verification becomes infeasible.

One possible approach is to introduce a parent managed account system:
    A parent account (email verified) could be used to create and manage one or more child accounts.
    Child accounts could have age-appropriate restrictions applied automatically such as chat filtering, time limits, or limited in-game features.
    The parent account would have access to settings and activity visibility for any linked child accounts.
    No email would be required for the child account itself only the parent’s email would be verified.

Overall though I'm not sure it makes sense to do unless financial transactions make it a liability. 

I think your approach of implementing proper self-protection tools like block/mute/etc.. commands and a blocklist for word pattern censorship will be highly effective.

**Comment by Volte6** [comment]
I agree, i think we can close this for now. Mute/deafen can be done on a case by case basis. I don't think children are accessing MUD's enough to care.
